addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -291,7 +291,7 @@ func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionRes
 		return gvr, nil
 	}
 	apiResourceList, err := wh.cachedDiscoveryClient.ServerPreferredResources()
-	if err != nil {
+	if _, ok := err.(*discovery.ErrGroupDiscoveryFailed); err != nil && !ok {
```

Comment:
if the error is ErrGroupDiscoveryFailed, we can safely continue, since an apiresult == nil will be catch later down the code and we will return the correct result if we don't find what we are looking for in the contents of apiresult

Reply:
Since you didn't check the error on line 293, wouldn't the following cast fail in case of err==nil?
```err.(*discovery.ErrGroupDiscoveryFailed)````

Reply:
no. I checked by running this code
```func main() {
	var err error
	if _, ok := err.(*discovery.ErrGroupDiscoveryFailed); !ok && err != nil {
		println("Error is not be ")
	}
	if err == nil {
		println("Error is nil")
	}
}

Reply:
Agreed, you are right, ```ok``` captures that case here